Wikipedia/Code/ReadingListsController.swift

```diff
@@ -491,6 +491,7 @@ public typealias ReadingListsController = WMFReadingListsController
         }
 
         newSyncState.remove(.needsUpdate)
+        WMFArticleSavedStateMigrationManager.shared.clearAll()
```

Comment:
Can we instead investigate and find where WMFArticle.savedDate is cleared out when this button is tapped? I would rather execute this clearing in the same spot in case there's something special about that placement that we aren't aware of.